gcloud/pubsub/topic.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Define API Topics."""
+
+import base64
+
+from gcloud.exceptions import NotFound
+
+
+class Topic(object):
+    """Topics are targets to which messages can be published.
+
+    Subscribers then receive those messages.
+
+    See:
+    https://cloud.google.com/pubsub/reference/rest/v1beta2/projects/topics
+
+    :type name: string
+    :param name: the name of the topic
+
+    :type project: string
+    :param project: the project to which the topic belongs.  If not passed,
+                    falls back to the default inferred from the environment.
+
+    :type connection: :class:gcloud.pubsub.connection.Connection
+    :param connection: the connection to use.  If not passed,
+                        falls back to the default inferred from the
+                        environment.
+    """
+    def __init__(self, name, project=None, connection=None):
```

Comment:
Hmm, I must be misunderstanding you here.  *Every* `Topic` API method has the project embedded in its URL.